banking/ebics/utils.py

```diff
@@ -332,6 +333,85 @@ def upload_camt_file():
 	process_camt_document(camt_document, bank_account)
 
 
+@frappe.whitelist()
+def upload_mt940_file():
+	frappe.has_permission("Bank Transaction", "create", throw=True)
+
+	file_bytes = frappe.local.uploaded_file
+	bank_account = frappe.form_dict.docname
+
+	register_fintech()
+
+	from fintech.swift import parse_mt940
+
+	mt940_data = file_bytes.decode()
+	statements: list[MT940Statement] = parse_mt940(mt940_data)
+
+	for statement in statements:
+		process_mt940_statement(statement, bank_account)
+
+
+def process_mt940_statement(statement: MT940Statement, bank_account: str):
+	"""Process a single MT940 statement and create bank transactions"""
+
+	company = frappe.db.get_value("Bank Account", bank_account, "company")
+	currency = statement["balance_open"]["currency"]
+	for transaction in statement["transactions"]:
+		create_mt940_bank_transaction(bank_account, company, transaction, currency)
+
+
+def create_mt940_bank_transaction(
+	bank_account: str, company: str, transaction: MT940Transaction, currency: str
+):
+	"""Create a bank transaction from MT940 transaction data"""
+
+	amount = transaction["amount"] or 0
+	description = transaction["description"] or transaction["sepa"]["SVWZ"] or ""
```

Comment:
```suggestion
	description = transaction["description"] or transaction.get("sepa", {}).get("SVWZ") or or " ".join(transaction.get("purpose", []))
```

In general, `dict.get(key, default_value)` handles missing keys gracefully while `dict[key]` fails.